realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -389,6 +389,22 @@ public Builder(Context context) {
             initializeBuilder(context.getFilesDir());
         }
 
+        /**
+         * Creates an instance of the Builder for the RealmConfiguration.
+         * <p>
+         * The Realm file will be saved in the provided folder, and it might require additional permissions.
+         *
+         * @param folder the folder to save Realm file in. Folder must be writable.
+         * @throws IllegalArgumentException if folder doesn't exist or isn't writable.
+         */
+        public Builder(Context context, File folder) {
```

Comment:
We should deprecate the old constructor when adding this. This will mean that in 2.0 we are 100% we have a Context which is useful for other API's (like `assetFile`).